core/state/database.go

```diff
@@ -223,7 +276,11 @@ func (db *CachingDB) ReadersWithCacheStats(stateRoot common.Hash) (ReaderWithSta
 
 // OpenTrie opens the main account trie at a specific root hash.
 func (db *CachingDB) OpenTrie(root common.Hash) (Trie, error) {
-	if db.triedb.IsVerkle() {
+	ts := db.LoadTransitionState(root)
```

Comment:
It's very expensive before the Verkle activation;

The TransitionState won't be existent before the verkle migration, each lookup ends up with an non-existent database lookup, which traverses the entire LSM Tree layers.

Reply:
this is the one I don't think we can replace with `db.triedb.IsVerkle()` because we won't be able to tell the difference between a transition and a post-transition. This being said, it might be enough to guard a call to `db.LoadTransitionState`...

The only part I'm not clear on, is how you would check that the db is loaded? don't you have to do some kind of empty marker reading either way?

Reply:
> this is the one I don't think we can replace with db.triedb.IsVerkle() because we won't be able to tell the difference between a transition and a post-transition. 

The idea for checking `db.triedb.IsVerkle` first is to skip the entire TransitionState checking before the verkle.
This state is essential to distinguish between a transition and a post-transition.